### PR TITLE
Wilmer -> feat: Add repository and extension method

### PR DESCRIPTION
**Pull request by Wil-JsDev**
### Repository

Add three repositories and the generic repository will have the methods to handle the database logic.

### Extension Method
This method works to handle dependency injections better.

### CODE REVIEW

File: services/OrderService/Repository/OrderItemRepository.cs [hunk 1]
```diff
@@ -1,37 +1,54 @@
...
-        throw new NotImplementedException();
+        _context.Remove(orderItemModel);
     }
 
-    public Task<bool> Exists(int id)
+    public async Task<bool> Exists(int id)
     {
-        throw new NotImplementedException();
+        var order = await _context.OrderItems.FirstAsync(x => x.Id == id);
+        return order == null ? true : false;
```
Comment (ManasesLovera) [inline services/OrderService/Repository/OrderItemRepository.cs]: it should be return order == null ? false : true; or order != null ? true : false. This is returning true if it's null which is the opposite it should do.

File: services/OrderService/Extensions/ServiceExtension.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+namespace OrderService.Extensions;
+
+public static class ServiceExtensio
```
Comment (ManasesLovera) [inline services/OrderService/Extensions/ServiceExtension.cs]: ServiceExtensions* with the 's'

File: services/OrderService/Repository/OrderRepository.cs [hunk 3]
```diff
@@ -1,37 +1,53 @@
...
-        throw new NotImplementedException();
+        _context.Remove(order);
     }
 
-    public Task<List<Order>> GetAllAsync()
+    public async Task<bool> Exists(int id)
     {
-        throw new NotImplementedException();
+        var order = await _context.Orders.FirstAsync(x => x.Id == id);
+        return order == null ? true : false;
```
Comment (ManasesLovera) [inline services/OrderService/Repository/OrderRepository.cs]: Looks like the same error as the previous one, please correct this is sending true when it doesn't exist and sending false when it does exist.

**Review by ManasesLovera — CHANGES_REQUESTED** [review]
I like the IGenericRepository, but please fix the issues I marked, after that please push the new commits and leave a comment that once you fixes this issue.

**Comment by Wil-JsDev** [comment]
Done!

**Review by ManasesLovera — APPROVED** [review]
Everything looks great now, good job!

**Pull request merged**